### PR TITLE
added ColorGradient

### PR DESCRIPTION
**Pull request by romulogcerqueira**
Created a customized, normalized and cross-graphical language heatmap. To generate the gradient, you just need to add the color points and the colors will be interpolated.

### CODE REVIEW

File: src/ColorGradientTypes.h [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#ifndef COLORGRADIENTTYPES_H
+#define COLORGRADIENTTYPES_H
+
+#include <iostream>
+
+namespace frame_helper {
+
+    enum ColormapType {
```
Comment (goldhoorn) [inline src/ColorGradientTypes.h]: This emun semms unused? You could mve this also in the jet map class

File: src/ColorGradientTypes.h [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#ifndef COLORGRADIENTTYPES_H
+#define COLORGRADIENTTYPES_H
+
+#include <iostream>
```
Comment (goldhoorn) [inline src/ColorGradientTypes.h]: This header is useless here

File: src/ColorGradient.h [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    struct ColorPoint {     // Internal class used to store colors at different points in the gradient.
+        float r, g, b;      // Red, green and blue values of our color.
+        float val;          // Position of our color along the gradient (between 0 and 1).
+        ColorPoint(float red, float green, float blue, float value) :
+                r(red), g(green), b(blue), val(value) {
+        }
+    };
+
+private:
+    std::vector<ColorPoint> color;      // An array of color points in ascending value.
```
Comment (goldhoorn) [inline src/ColorGradient.h]: A list would make more sense you inserting elements in the middle and have no random access a list for s more efficient then

File: src/ColorGradientTypes.h (hunk 1 above)
Comment (romulogcerqueira) [inline src/ColorGradientTypes.h]: Done!

File: src/ColorGradientTypes.h (hunk 2 above)
Comment (romulogcerqueira) [inline src/ColorGradientTypes.h]: Done!

File: src/ColorGradient.h (hunk 3 above)
Comment (romulogcerqueira) [inline src/ColorGradient.h]: I agree with you. Inserting into the middle of std::vector requires all elements after the insertion point are shuffled, while on std::list requires just to create a new node, store the value and connect the pointers. Done!

**Comment by goldhoorn** [comment]
Thaks for beeing patient and fullfilling all comments :+1:

**Pull request merged**